pkg/minishift/addon/command/openshift_command.go

```diff
@@ -36,13 +36,13 @@ func NewOpenShiftCommand(command string, ignoreError bool) *OpenShiftCommand {
 }
 
 func (c *OpenShiftCommand) doExecute(ec *ExecutionContext, ignoreError bool) error {
-	// split off the actual 'oc' command. We are using our cached oc version to run oc commands
```

Comment:
Why does it show `our cached oc version` but we actually do not run it as such?

Reply:
@gbraad it was old comment which I think during implementation hardy copy pasted same text from oc commander.

Reply:
this code needs a quick review overall...

but to continue, let's split out the preconditions to when we can run `openshift` or `oc`, or does running `oc` always works?